### PR TITLE
Add interop example for `arrow::StringViewArray` to `cudf::column`

### PR DESCRIPTION
**Pull request by JayjeetAtGithub**
## Description
Demonstrates the conversion from an `arrow:StringViewArray` to a `cudf::column`

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/examples/interop/interop.cu
Comment (davidwendt) [inline cpp/examples/interop/interop.cu]: Can we remove the thrust device code dependency? I don't think it adds to the example and may be more approachable as a .cpp file. The copy function here can be replaced with a `cudaMemcpyAsync` and the inclusive-scan can be done in host code using [`std::inclusive_scan`](https://en.cppreference.com/w/cpp/algorithm/inclusive_scan) and then the results copied to device memory using `cudaMemcpyAsync`.

File: cpp/examples/interop/interop.cu
Comment (davidwendt) [inline cpp/examples/interop/interop.cu:147]: I think you can use `exclusive-scan` instead of `inclusive-scan` to avoid the extra call to set the first element to 0 and the extra copy of the sizes. Also, see my other comment about doing this in host code instead and just copying the offsets to device memory at the end.

**Review by davidwendt — CHANGES_REQUESTED** [review]

File: cpp/examples/interop/interop.cpp
Comment (davidwendt) [inline cpp/examples/interop/interop.cpp:92]: Looks like you copied this from the `strings_column_wrapper` code. You'll need a different exception message.

File: cpp/examples/interop/interop.cpp
Comment (davidwendt) [inline cpp/examples/interop/interop.cpp:147]: Could we make more strings out of these? Also is `nvidia` added twice?
```suggestion
  views.push_back(ToStringView("hello, 0, 0));
  views.push_back(ToStringView("world", 0, 5));
  views.push_back(ToStringView("apache", 0, 10));
  views.push_back(ToStringView("arrow", 0, 16));
  views.push_back(ToStringView("cudf", 0, 21));
  views.push_back(ToInlineStringView("nvidia"));
```

File: cpp/examples/interop/interop.cpp
Comment (davidwendt) [inline cpp/examples/interop/interop.cpp]: I wonder if we could do some operation here. Perhaps a [reduce](https://docs.rapids.ai/api/libcudf/stable/group__aggregation__reduction#ga6e7a59da36673a0d50cb26a471a70d53) or [get_element](https://docs.rapids.ai/api/libcudf/stable/group__column__copy.html#gabcecf3e790a7dd4800f99e61e8e93927).
Both return a [`string_scalar`](https://docs.rapids.ai/api/libcudf/stable/classcudf_1_1string__scalar.html) which has a [`to_string()`](https://docs.rapids.ai/api/libcudf/stable/classcudf_1_1string__scalar#a0e17858682c82a055693964f633b3444) function that will copy the string to the host so you can print it to the console.

**Review by davidwendt — CHANGES_REQUESTED** [review]

File: cpp/examples/interop/interop.cpp
Comment (davidwendt) [reply cpp/examples/interop/interop.cpp]: Looks like you changed this to output as a CSV file. That is ok too.

File: cpp/examples/interop/interop.cpp
Comment (JayjeetAtGithub) [reply cpp/examples/interop/interop.cpp:92]: I think we can remove the error message here

Comment (JayjeetAtGithub) [reply cpp/examples/interop/interop.cpp]: Nevermind, let's keep it.

File: cpp/examples/interop/interop.cpp
Comment (davidwendt) [reply cpp/examples/interop/interop.cpp:147]: I'd like the strings to be 'readable' so make them phrases or sentences (i.e. add spaces) if you want to make them longer.

File: cpp/examples/interop/interop.cpp
Comment (davidwendt) [inline cpp/examples/interop/interop.cpp:78]:
```suggestion
 * constituent chars and a vector of offsets.
```
The wording is confusing here I think.

File: cpp/examples/interop/interop.cpp
Comment (davidwendt) [inline cpp/examples/interop/interop.cpp:145]:
```suggestion
  auto buffer_a = arrow::Buffer::FromString("hello rapids teamapache arrow interopnvidiacudf");
  data_buffers.push_back(buffer_a);
  views.push_back(ToStringView("hello rapid steam", 0, 0));
  views.push_back(ToStringView("apache arrow interop", 0, 17));
```

File: cpp/examples/interop/interop.cpp
Comment (ttnghia) [inline cpp/examples/interop/interop.cpp:18]: These headers are not in the same folder as this source file thus they should be wrapped in `<>`.

File: cpp/examples/interop/interop.cpp
Comment (ttnghia) [inline cpp/examples/interop/interop.cpp:27]: All the function names should follow cudf convention.
```suggestion
inline arrow::BinaryViewType::c_type to_inline_string_view(const void* data, int32_t size)
```

File: cpp/examples/interop/interop.cpp
Comment (ttnghia) [inline cpp/examples/interop/interop.cpp:68]: Use `auto const` whenever possible.

File: cpp/examples/interop/interop.cpp
Comment (ttnghia) [inline cpp/examples/interop/interop.cpp:31]: I don't know the details of the `inlined` constructor, but I guess this initializes it to have an empty data array? If so, will the `memcpy` statement write the input data into an empty array?

File: cpp/examples/interop/interop.cpp
Comment (ttnghia) [inline cpp/examples/interop/interop.cpp:104]: Snake case please.

File: cpp/examples/interop/interop.cpp
Comment (ttnghia) [inline cpp/examples/interop/interop.cpp:63]: Snake case please.

File: cpp/examples/interop/interop.cpp
Comment (JayjeetAtGithub) [reply cpp/examples/interop/interop.cpp:31]: If we look into the `arrow::BinaryViewType::c_type` struct [here](https://github.com/apache/arrow/blob/main/cpp/src/arrow/type.h#L823), we see that the `data` member of the `inlined` struct is a pre-allocated `std::array` of size `kInlineSize`.  Also, the `prefix` member of the `ref` struct (for non-inline views), is a pre-allocated `std::array` of size `kPrefixSize`.  Does that answer your question ?

**Review by ttnghia — APPROVED** [review]

**Comment by davidwendt** [comment]
@JayjeetAtGithub We generally require 2 C++ review approvals before merging a PR.

**Comment by GregoryKimball** [comment]
@karthikeyann I realize this PR already merged, but would you please take a look over the example and suggest changes?

**Pull request merged**
